frontend/lib/src/components/widgets/DataFrame/hooks/useDataLoader.ts

```diff
@@ -73,7 +73,16 @@ function useDataLoader(
           originalRow
         )
         if (notNullOrUndefined(editedCell)) {
-          return editedCell
+          // Create a new representation of the edited cell to apply
+          // changes that might have been applied to the column (e.g. change of format from UI).
+          // TODO(lukasmasuch): We should refactor this at some point to avoid storing
+          // cells in the editing state. It would be enough to store the value and the
+          // last updated timestamp.
```

Comment:
I added a tech debt entry for this [here](https://www.notion.so/snowflake-corp/Refactor-data-editor-editing-state-to-only-store-raw-values-and-not-the-cell-object-1a17170bb416805bbf17fd49abccedf9).